src/mbgl/map/live_tile_data.cpp

```diff
@@ -1,58 +1,57 @@
 #include <mbgl/map/annotation.hpp>
 #include <mbgl/map/live_tile_data.hpp>
 #include <mbgl/map/live_tile.hpp>
-#include <mbgl/map/tile_parser.hpp>
-#include <mbgl/map/source.hpp>
-#include <mbgl/map/vector_tile.hpp>
-#include <mbgl/platform/log.hpp>
+#include <mbgl/util/worker.hpp>
+#include <mbgl/util/work_request.hpp>
 
 using namespace mbgl;
 
 LiveTileData::LiveTileData(const TileID& id_,
                            AnnotationManager& annotationManager_,
                            Style& style_,
-                           GlyphAtlas& glyphAtlas_,
-                           GlyphStore& glyphStore_,
-                           SpriteAtlas& spriteAtlas_,
-                           util::ptr<Sprite> sprite_,
                            const SourceInfo& source_,
                            float angle_,
                            bool collisionDebug_)
-    : VectorTileData::VectorTileData(id_, style_, glyphAtlas_, glyphStore_,
-                                     spriteAtlas_, sprite_, source_, angle_, collisionDebug_),
+    : VectorTileData(id_, style_, source_, angle_, collisionDebug_),
       annotationManager(annotationManager_) {
-    // live features are always ready
-    setState(State::loaded);
+    // live features are always loaded
+    state = State::loaded;
 }
 
 LiveTileData::~LiveTileData() {
-    // Cancel in most derived class destructor so that worker tasks are joined before
```

Comment:
My eventual plan was to have a flat hierarchy, with `LiveTileData` _not_ inheriting from `VectorTileData`, so that (among other things) this would become moot. If we end up doing that I'll add the comments back.